src/types.c

```diff
@@ -133,6 +136,11 @@ bool is_number_type(const Type *t)
 {
     return is_integer_type(t) || t->kind == TYPE_DOUBLE;
 }
+
+bool is_float_type(const Type *t)
+{
+    return is_integer_type(t) || t->kind == TYPE_FLOAT;

```

Comment:
I think this is causing a lot of test failures. It's a function that is more confusing than helpful, and basically impossible to use in a way that makes sense: `is_float_type(intType)` returns true.